docker-app/qfieldcloud/core/utils.py

```diff
@@ -296,3 +308,69 @@ def get_s3_object_url(
         str: URL
     """
     return f"{settings.STORAGE_ENDPOINT_URL_EXTERNAL}/{bucket.name}/{key}"
+
+
+def list_versions(
+    bucket: mypy_boto3_s3.service_resource.Bucket,
+    prefix: str,
+    strip_prefix: bool = True,
+) -> Iterable[S3ObjectVersion]:
+    """Iterator that lists a bucket's objects under prefix."""
+    for v in bucket.object_versions.filter(Prefix=prefix):
+        if strip_prefix:
+            start_idx = len(prefix)
+            name = v.key[start_idx:]
+        else:
+            name = v.key
+
+        yield S3ObjectVersion(name, v)
+
+
+def get_prev_string_for_sorting(s: str) -> str:
+    """Get a string that is one string before the given string, if sorted alphabetically."""
+    if len(s) == 0:
+        return s
+
+    s, c = s[:-1], ord(s[-1])
+    if c > 0:
+        s += chr(c - 1)
+    s += "".join(["\u7FFF" for _ in range(10)])
+    return s
```

Comment:
I don't see it being used, an not really clear what this does.

Or is that a test to see if I actually review the code ? :-D

Reply:
Haha, got me! 

Well, this was a artefact from the first implementation I used, that was not asuming the files are ordered. Since I realized that S3 always returns the object keys sorted, so there is no need for this magic anymore.

Reply:
Will remoe it, thanks.